cumulus/parachains/common/src/xcm_config.rs

```diff
@@ -78,3 +78,59 @@ impl<Location: Get<MultiLocation>> ContainsPair<MultiAsset, MultiLocation>
 		matches!(asset.id, Concrete(ref id) if id == origin && origin == &Location::get())
 	}
 }
+
+/// Accepts an asset if it is a native asset from the system (Relay Chain or system parachain).
+pub struct ConcreteNativeAssetFromSystem;
+impl ContainsPair<MultiAsset, MultiLocation> for ConcreteNativeAssetFromSystem {
+	fn contains(asset: &MultiAsset, origin: &MultiLocation) -> bool {
+		log::trace!(target: "xcm::contains", "ConcreteNativeAssetFromSystem asset: {:?}, origin: {:?}", asset, origin);
+		let parent = MultiLocation::parent();
+		let is_system = match origin {
+			// The Relay Chain
+			MultiLocation { parents: 1, interior: Here } => true,
+			// System parachain
+			MultiLocation { parents: 1, interior: X1(Parachain(id)) } if *id < 2000 => true,
+			// Others
+			_ => false,
+		};
+		matches!(asset.id, Concrete(id) if id == parent && is_system)
```

Comment:
Checking `id == parent` here means that you only check whether the asset is the relay chain native token (assuming that the origin is a parachain), which seems incorrect given the description of this type.

Reply:
Ah, all system parachains use the relay chain native token as the currency, so this is fine.